pkg/scsi/spc.go

```diff
@@ -227,30 +290,43 @@ func SPCReportLuns(host int, cmd *api.SCSICommand) api.SAMStat {
 		glog.Warningf("goto sense, allocationLength < 16")
 		goto sense
 	}
-	remainLength = allocationLength - 8
-	availLength = 8 * uint32(len(cmd.Target.Devices))
+
+	remainLength = allocationLength
+	if cmd.Target.Devices[0] == nil {
+		availLength = 8 * uint32(len(cmd.Target.Devices)+1)
+	} else {
+		availLength = 8 * uint32(len(cmd.Target.Devices))
+	}
+
+	// LUN list length
 	buf.Write(util.MarshalUint32(availLength))
 	cmd.InSDBBuffer.Resid = int32(actualLength)
+
 	// Skip through to byte 8, Reserved
 	for i := 0; i < 4; i++ {
 		buf.WriteByte(0x00)
 	}
 
-	for lunumber, lu := range cmd.Target.Devices {
-		glog.V(2).Infof("LUN: ", lunumber)
+	//For LUN0
+	if cmd.Target.Devices[0] == nil {
```

Comment:
Dummy LUN0 isn't in target.Devices. Target.Devices[0] == nil means we are using dummy LUN0 instead of a real LUN. We should put lun0 into the lun list or the loop cannot help us to do it.